src/transformers/models/mplug_owl/processing_mplug_owl.py

```diff
@@ -0,0 +1,217 @@
+    r"""
+    A warp of CLIPImageProcessor
+    """
+    pass
+
+
+def detokenize_generations(tokens_gpu_tensor, lengths_gpu_tensor, return_segments, tokenizer):
+    """Detokenize the generated tokens."""
+
+    prompts_plus_generations = []
+    if return_segments:
+        prompts_plus_generations_segments = []
+
+    tokens = tokens_gpu_tensor.cpu().numpy().tolist()
+    lengths = lengths_gpu_tensor.cpu().numpy().tolist()
+    for sequence_tokens, length in zip(tokens, lengths):
+        sequence_tokens = sequence_tokens[:length]
+        prompts_plus_generations.append(tokenizer.detokenize(sequence_tokens))
+        if return_segments:
+            from tokenizers.decoders import Metaspace
+
+            if hasattr(tokenizer, "tokenizer"):
+                if isinstance(tokenizer.tokenizer.decoder, Metaspace):
+                    words = tokenizer.tokenizer.decode(sequence_tokens)
+                else:
+                    words = []
+                    for token in sequence_tokens:
+                        word = tokenizer.tokenizer.decoder[token]
+                        word = bytearray([tokenizer.tokenizer.byte_decoder[c] for c in word]).decode(
+                            "utf-8", errors="replace"
+                        )
+                        words.append(word)
+                prompts_plus_generations_segments.append(words)
+            else:
+                words = tokenizer.detokenize(sequence_tokens)
+                # else:
+                #     words = []
+                #     for token in sequence_tokens:
+                #         word = tokenizer.tokenizer.decoder[token]
+                #         word = bytearray(
+                #             [tokenizer.tokenizer.byte_decoder[c] for c in word]).decode(
+                #                 'utf-8', errors='replace')
+                #         words.append(word)
+                prompts_plus_generations_segments.append(words)
+
+    if return_segments:
+        return tokens, prompts_plus_generations, prompts_plus_generations_segments
+
+    return tokens, prompts_plus_generations
+
+
+def tokenize_prompts(prompts=None, tokens_to_generate=None, add_BOS=None, rank=0, tokenizer=None, **kwargs):
+    """Tokenize prompts and make them avaiable on all ranks."""
+
+    # On all ranks set to None so we can pass them to functions
+    prompts_tokens_cuda_long_tensor = None
+    prompts_length_cuda_long_tensor = None
+
+    # On the specified rank, build the above.
+    attention_mask = None
+
+    prompts_tokens_cuda_long_tensor, prompts_length_cuda_long_tensor, attention_mask = _tokenize_prompts_and_batch(
+        prompts, tokens_to_generate, add_BOS, tokenizer, **kwargs
+    )
+
+    return {
+        "input_ids": prompts_tokens_cuda_long_tensor,
+        "attention_mask": attention_mask,
+        # "prompt_length": prompts_length_cuda_long_tensor,
+    }
+
+
+def _tokenize_prompts_and_batch(prompts, tokens_to_generate, add_BOS, tokenizer, **kwargs):
+    import torch
+
```

Comment:
I would probably move these methods as private methods of the processor
Could you also elaborate on why you can't natively use the `tokenizer`'s call method directly ?

Reply:
To accommodate the interleaved multi-modal scenario (for example, text image text image), we propose tokenizing the placeholder word ```<image>``` into 65 tokens (Then we can substituting their embedding with features extracted from visual encoder and pass them to llama during forward). However, we are unsure how to using the native tokenizer function to convert a word into 65 tokens.